tests/test_interconnect/test_interconnect_cgra.py

```diff
@@ -59,6 +59,7 @@ def test_interconnect_point_wise(batch_size: int, dw_files, io_sides):
     add_bs = tile.core.get_config_bitstream(asm.umult0())
     for addr, data in add_bs:
         config_data.append((interconnect.get_config_addr(addr, 0, x, y), data))
+    config_data = compress_config_data(config_data)
```

Comment:
should we have separate set of tests that compress config data vs. not? Just so we know both are working. We could just parameterize the tests on flag that does compression vs. not

Reply:
GIven the current design, you have to merge/compress the config data, otherwise it won't work. Say you have two config
```
00000001 100000000
00000001 010000000
```
Since all the config registers are packed together, you really need this to configure the register
```
00000001 11000000
```

Reply:
Ah I see, I thought there would be 2 modes. it would be good to think about how to fold that into the bitstream generation itself. I imagine an annoying bug would be forgetting to call `compress_config_data`.

One idea would be to make the bitstream a special class with methods rather than a plain list of tuples. That way it would be "automatically" compressed. Or if the `get_config_bitstream` methods did the compression themselves before returning. This seems safer and not so intrusive.

Reply:
I can work on a new bitstream class the handle that. This will be in the future PR.

Reply:
Sounds good.